packages/shared/lib/services/connection.service.ts

```diff
@@ -900,15 +902,13 @@ class ConnectionService {
             action: 'token_refresh' | 'connection_test';
         }) => Promise<void>;
         connectionTestHook?:
-            | ((
-                  providerName: string,
-                  provider: Provider,
-                  credentials: ApiKeyCredentials | BasicApiCredentials | TbaCredentials,
-                  connectionId: string,
-                  providerConfigKey: string,
-                  environment_id: number,
-                  connection_config: ConnectionConfig
-              ) => Promise<Result<boolean, NangoError>>)
+            | ((args: {
+                  config: Config;
+                  provider: Provider;
+                  credentials: ApiKeyCredentials | BasicApiCredentials | TbaCredentials | JwtCredentials | SignatureCredentials;
+                  connectionId: string;
+                  connectionConfig: ConnectionConfig;
+              }) => Promise<Result<{ logs: MessageRowInsert[] }, NangoError>>)
```

Comment:
maybe `connectionTestHook` deserves its own named type. wdyt?

Reply:
I think the main issue is that hooks are in the server and this specific thing is in shared, wdyt about moving this specific code to server? 
Alternatively we can move all hooks to shared

Reply:
Originally hooks [were in shared](https://github.com/NangoHQ/nango/pull/1407/files#diff-bea6e4872550e34e4234c63f78a893adabfceb768ca41ad82a520dc3edddd591) but @TBonnin [moved them to shared](https://github.com/NangoHQ/nango/pull/2188) actually.

Reply:
arg good catch thanks 👍🏻 
I'll wait for thomas opinion then :D

Reply:
My original comment was just about the type so it is less verbose, not moving the hooks logic. 😄 

But if we want to talk about dependency management, the main challenge and question is: how to we want to inject dependencies:
1. hooks outside of shared and inject hook functions into shared that encapsulate their own dependencies (shared doesn't have to care)
2. hooks inside share and inject all the dependencies the hooks needs (shared needs to know about the deps)

i though the first one was cleaner. For instance we don't follow this pattern [here](https://github.com/NangoHQ/nango/blob/72ca1714f9e4e943955550f0d59db4d175592797/packages/shared/lib/services/connection.service.ts#L879-L881). As a result we have to pass the `orchestrator` and `logContextGetter` just because it is required by the side effects executed after a connection is deleted. Instead we could pass a `postConnectionDeletion` hook that is opaque for shared who doesn't need to know anymore about the orchestrator

Reply:
Yeah okay I see, I talked about moving code because dissociating the types and the code is always painful. 

---

> i though the first one was cleaner. For instance we don't follow this pattern [here](https://github.com/NangoHQ/nango/blob/72ca1714f9e4e943955550f0d59db4d175592797/packages/shared/lib/services/connection.service.ts#L879-L881). As a result we have to pass the orchestrator and logContextGetter just because it is required by the side effects executed after a connection is deleted.

We mostly did that because of the CLI though not because we really wanted to, at least I wouldn't have bothered with logs.

Opinion: We moved stuff outside shared for multiple reasons, mostly CLI and code cleaning. Once we have cut CLI from shared, there will be fewer reasons to move things outside `shared` things like this code: not big enough to be a package but still shared in various parts of the code. 
We can still do dependency injection even if the code lives inside shared (handy for testing). 
I'll be in favor of actually moving this back since it's reused. Then the way we used is up to the person.

I'll go ahead and merge then

Reply:
> We mostly did that because of the CLI though not because we really wanted to, at least I wouldn't have bothered with logs.

True and for logs it would make kind of sense. Not sure about the orchestrator though. Anyway it is tangential to this PR so go ahead and merge